qiita_db/data.py

```diff
@@ -1173,3 +1332,39 @@ def processed_date(self):
         return conn_handler.execute_fetchone(
             "SELECT processed_date FROM qiita.{0} WHERE "
             "processed_data_id=%s".format(self._table), (self.id,))[0]
+
+    @property
+    def status(self):
+        conn_handler = SQLConnectionHandler()
+        sql = """SELECT processed_data_status
+                FROM qiita.processed_data_status pds
+                  JOIN qiita.processed_data pd
+                    ON pd.processed_data_status_id=pds.processed_data_status_id
+                WHERE pd.processed_data_id=%s"""
+        return conn_handler.execute_fetchone(sql, (self._id,))[0]
+
+    @status.setter
+    def status(self, status):
+        """Set the status value
+
+        Parameters
+        ----------
+        status : str
+            The new status
+
+        Raises
+        ------
+        QiitaDBStatusError
+            If the processed data status is public
+        """
+        if self.status == 'public':
+            raise QiitaDBStatusError(
+                "Illegal operation on public processed data")
+
+        conn_handler = SQLConnectionHandler()
+        sql = """UPDATE qiita.{0} SET processed_data_status_id = (
```

Comment:
Note that these objects do not extend the QiitaObjectStatus class, they extend the QiitaObject class